Docs: Add caching support to Go multi-Arch examples (#5138)
dagger/dagger: docs/current/guides/406009-multiplatform-support.md

The final image is still multi-platform because each `Container` set as a `PlatformVariant` was initialized with a specific platform (after the cross-compilation has occurred, at the bottom of the `for` loop in the code above).

> ### Adding Caching Support

Reviewer: ```suggestion
### Add caching
```

> Caching can significantly improve the speed of your builds in Dagger by saving and reusing the results of expensive operations, such as downloading dependencies or compiling code. Dagger supports caching via the `withMountedCache` function, which mounts a cache directory from the host machine into the container.

Reviewer: ```suggestion
Caching can significantly improve the speed of your builds in Dagger by saving and reusing the results of expensive operations, such as downloading dependencies or compiling code. Dagger supports caching via the `withMountedCache()` function, which mounts a cache directory from the host machine into the container.
```

> We will compile the Go project twice. The first compilation will take longer because the cache directories are empty. The second compilation should be faster, as the cache directories will contain the necessary Go modules and build outputs from the first compilation:

Reviewer: ```suggestion
The following code listing compiles the Go project twice. The first compilation will take longer because the cache directories are empty. The second compilation should be faster, as the cache directories will contain the necessary Go modules and build outputs from the first compilation:
```

```go file=./snippets/multiplatform-support/cache/main.go

```

> In this example, we define a build function that sets up the container, mounts the cache directories, and compiles the Go project. We run this function twice and measure the build time for each run. The second build should be faster, demonstrating the benefits of using cache mounts.

Reviewer: ```suggestion
This example defines a `build()` function that sets up the container, mounts the cache directories, and compiles the Go project. The function runs twice, and the build time for each run is measured. The second build is faster, demonstrating the benefits of using cache mounts.
```
